custom_completions.go

```diff
@@ -271,20 +272,24 @@ func (c *Command) getCompletions(args []string) (*Command, []string, ShellCompDi
 	var completions []string
 	directive := ShellCompDirectiveDefault
 	if flag == nil {
-		// Check if there are any local, non-persistent flags on the command-line
 		foundLocalNonPersistentFlag := false
-		localNonPersistentFlags := finalCmd.LocalNonPersistentFlags()
-		finalCmd.NonInheritedFlags().VisitAll(func(flag *pflag.Flag) {
-			if localNonPersistentFlags.Lookup(flag.Name) != nil && flag.Changed {
-				foundLocalNonPersistentFlag = true
-			}
-		})
+		// If TraverseChildren is true we dont check for local flags
+		// because we can use a local flag on a parent command
+		if !finalCmd.TraverseChildren {
```

Comment:
From what I understand, for `TraverseChildren` to work (outside of completion), it *must* be set on the Root command.  This is because `TraverseChildren` is only considered here:
https://github.com/spf13/cobra/blob/19e41cf081df9bf9dde5cffa0090e718c3fdc8af/command.go#L922
and that line is *only* run on the root command.
I have confirmed by testing that setting `TraverseChildren` only on a sub-command will still cause an "unknow flag" error for local flags.

Therefore, we should only consider `TraverseChildren` for the root command, so it should probably be:
`if !finalCmd.Root().TraverseChildren {`

Reply:
Yes you are right, I had a wrong understanding where TraverseChildren should be set. I updated the code according to your notes.